### PR TITLE
Feat/clickable layers

### PR DESCRIPTION
**Pull request by bob-voorhoede**
- `clickable` prop for layer, to make a layer emit a click event when it is clicked
- mouse pointer change when hovering over clickable layer

### CODE REVIEW

File: src/components/v-mapbox-layer.js [hunk 1]
```diff
@@ -63,10 +75,29 @@ export default {
...
       }
     },
 
+    clickFn(e) {
+      this.$emit('click', e);
+    },
+
+    mouseEnterFn() {
+      const map = this.getMap();
+      map.getCanvas().style.cursor = 'pointer';
```
Comment (CindyvdVries) [inline src/components/v-mapbox-layer.js:93]: just an idea; maybe cursor style should be a prop the user can specify?

File: src/stories/index.stories.js [hunk 2]
```diff
@@ -178,6 +178,17 @@ const dynamicLayersTemplate = `
...
 `
 
+const clickableLayersTemplate = `
+<v-mapbox
+ :map-style="style"
+ access-token="pk.eyJ1IjoibWFwYm94IiwiYSI6ImNpejY4NXVycTA2emYycXBndHRqcmZ3N3gifQ.rJcFIG214AriISLbB6B5aw"
+ style="height: 300px;"
+ :center="center"
+>
+  <v-mapbox-layer v-for="layer in layers" :options="layer" :key="layer.id" :clickable="true" @click="onLayerClick" />
```
Comment (CindyvdVries) [inline src/stories/index.stories.js:188]: if you have multiple layers that are clickable and overlapping, does this still work properly??

Comment (CindyvdVries) [inline src/stories/index.stories.js:188]: if you have a raster layer, it cannot be clickable in this way?

**Review by CindyvdVries — APPROVED** [review]

File: src/stories/index.stories.js (hunk 2 above)
Comment (bob-voorhoede) [reply src/stories/index.stories.js:188]: you click the one that's on top

File: src/components/v-mapbox-layer.js (hunk 1 above)
Comment (bob-voorhoede) [reply src/components/v-mapbox-layer.js:93]: may be we can handle this as a separate feature request ☺️

File: src/stories/index.stories.js (hunk 2 above)
Comment (bob-voorhoede) [reply src/stories/index.stories.js:188]: that's true, they are never clickable. will put it in stories so devs know about this limitation

**Pull request merged**